packages/dynamic_forms/lib/src/form_elements/form_element.dart

```diff
@@ -35,6 +42,30 @@ abstract class FormElement implements ExpressionProviderElement {
     throw Exception("Can't get expressions for $propertyName");
   }
 
+  @override
+  ExpressionProviderElement clone(
+      ExpressionProvider<ExpressionProviderElement> parent) {
+    var result = getInstance();
+    result._fillFromDictionary(this, result, parent);
+    result.fillFormElement(
+        id: this.id, parent: parent, isVisible: this.isVisible);
+    return result;
+  }
+
+  void _fillFromDictionary(
+      FormElement formElement,
+      ExpressionProviderElement instance,
+      ExpressionProvider<ExpressionProviderElement> parent) {
+    var formElementProperties = formElement.getProperties();
+    formElementProperties.forEach((k, v) {
+      if (v is List) properties[k] = cloneChildren(v, instance);
+      if ((this is RequiredValidation) && (k == IS_VALID_PROPERTY_NAME))
```

Comment:
I wouldn't reference children FormElements in clone method -> RequiredValidation is one exception from the clone rule, but there may be more. One solution would be to let each child which requires special clone logic to completely override clone method, but this is too much work. 

Instead I would suggest to create new method, which child can hook on and override it. In this case I would extract the whole block inside the forEach to the protected method called for example `cloneProperty(String key, ElementValue oldProperty, ExpressionProvider<ExpressionProviderElement> parent)`... And the RequiredValidation would override this method and will check if the key is IS_VALID_PROPERTY_NAME and in this case would do the special action. In all other cases would call `super.cloneProperty`....